bundles/org.openhab.binding.mercedesme/src/main/java/org/openhab/binding/mercedesme/internal/api/Authorization.java

```diff
@@ -0,0 +1,533 @@
+     * @throws MercedesMeApiException if an error occurs during API call
+     */
+    private String getResumeUrl(HttpClient loginHttpClient, String codeChallenge)
+            throws MercedesMeAuthException, MercedesMeApiException {
+        Fields resumeContent = new Fields();
+        resumeContent.put("client_id", Constants.AUTH_CLIENT_ID);
+        resumeContent.put("code_challenge_method", "S256");
+        resumeContent.put("redirect_uri", Constants.AUTH_REDIRECT_URI);
+        resumeContent.put("response_type", "code");
+        resumeContent.put("scope", Constants.AUTH_SCOPE);
+        resumeContent.put("code_challenge", codeChallenge);
+
+        String resumeUrl = null;
+        Request resumeRequest = loginHttpClient
+                .newRequest(baseUrl + "/as/authorization.oauth2?" + FormContentProvider.convert(resumeContent))
+                .followRedirects(true);
+        resumeRequest.header(HttpHeader.USER_AGENT, Constants.AUTH_USER_AGENT);
+        resumeRequest.header(HttpHeader.ACCEPT_LANGUAGE, Constants.AUTH_LANGUAGE);
+        resumeRequest.header(HttpHeader.ACCEPT, "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8");
+        ContentResponse resumeResponse = send(resumeRequest);
+        logger.trace("Step 1: Get resume code {} - {}", resumeResponse.getStatus(),
+                resumeResponse.getRequest().getURI());
+        if (resumeResponse.getStatus() == 200) {
+            String response = resumeResponse.getRequest().getURI().getQuery();
+            Map<String, String> params = Utils.getQueryParams(response);
+            resumeUrl = params.get("resume");
+        }
+        if (resumeUrl != null) {
+            return resumeUrl;
+        } else {
+            throw new MercedesMeAuthException("Failed to get resume URL. HTTP " + resumeResponse.getStatus());
+        }
+    }
+
+    /**
+     * Send user agent.
+     *
+     * @throws MercedesMeAuthException if response status isn't correct
+     * @throws MercedesMeApiException if an error occurs during API call
+     */
+    private void sendUserAgent(HttpClient loginHttpClient) throws MercedesMeAuthException, MercedesMeApiException {
+        Request agentRequest = loginHttpClient.POST(baseUrl + "/ciam/auth/ua");
+        agentRequest.header(HttpHeader.USER_AGENT, Constants.AUTH_USER_AGENT);
+        agentRequest.header(HttpHeader.ACCEPT_LANGUAGE, Constants.AUTH_LANGUAGE);
+        agentRequest.header(HttpHeader.ACCEPT, "*/*");
+        agentRequest.header(HttpHeader.CONTENT_TYPE, CONTENT_TYPE_JSON);
+        agentRequest.header(HttpHeader.ORIGIN, baseUrl);
+
+        JSONObject agentContent = new JSONObject();
+        agentContent.put("browserName", "Mobile Safari");
+        agentContent.put("browserVersion", "15.6.6");
+        agentContent.put("osName", "iOS");
+        agentRequest.content(new StringContentProvider(agentContent.toString(), "utf-8"));
```

Comment:
See comment further down.
```suggestion
        agentRequest.agent(Constants.AUTH_USER_AGENT);
        agentRequest.header(HttpHeader.ACCEPT_LANGUAGE, Constants.AUTH_LANGUAGE);
        agentRequest.header(HttpHeader.ACCEPT, "*/*");
        agentRequest.header(HttpHeader.ORIGIN, baseUrl);

        JSONObject agentContent = new JSONObject();
        agentContent.put("browserName", "Mobile Safari");
        agentContent.put("browserVersion", "15.6.6");
        agentContent.put("osName", "iOS");
        agentRequest.content(new StringContentProvider(agentContent.toString(), "utf-8"), CONTENT_TYPE_JSON);
```